server/auth/store.go

```diff
@@ -571,7 +577,11 @@ func (as *authStore) UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse,
 }
 
 func (as *authStore) UserList(r *pb.AuthUserListRequest) (*pb.AuthUserListResponse, error) {
-	users := as.be.GetAllUsers()
+	tx := as.be.ReadTx()
+	tx.Lock()
+	defer tx.Unlock()
+
+	users := tx.UnsafeGetAllUsers()
```

Comment:
```suggestion
users := as.be.GetAllUsers()
```

I should add this comment earlier, sorry for the confusion. The readTx indeed can't see the changes in the batchTx before being committed.